dev/AppNotifications/AppNotificationManager.cpp

```diff
@@ -167,6 +168,28 @@ namespace winrt::Microsoft::Windows::AppNotifications::implementation
         }
     }
 
+    void AppNotificationManager::Register()
+    {
+        RegisterHelper(winrt::hstring{} /* displayName */, std::wstring{} /* iconFilePath */);
+    }
+
+    void AppNotificationManager::Register(hstring const& displayName, winrt::Uri const& iconUri)
+    {
+        THROW_HR_IF_MSG(E_ILLEGAL_METHOD_CALL, AppModel::Identity::IsPackagedProcess(), "Not applicable for packaged applications");
+
+        THROW_HR_IF(E_INVALIDARG, (displayName != winrt::hstring{}) xor (iconUri != nullptr));
```

Comment:
xor? Can we use && or || :D?

Reply:
If any of the arguments are empty, we should throw.

Reply:
I chose xor here because it serves the purpose:

Below are the two combinations of the displayname and icon the xor operation would allow and not throw:
Scenario 1: If (displayName = winrt::hstring{}  &&  iconUri = nullptr) -  In this scenario the windowsappsdk will get the icon from .exe

Scenario 2: If (displayName != winrt::hstring{} && iconUri != nullptr) - In this scenario we pick the developer provided parameters.

Note: winrt::hstring{} means basically setting the parameter to nullptr.

Thanks.

Reply:
Now once the xor operation allows the right combinations, only when the iconUri != nullptr we check if path exists.

Reply:
I thought we wanted this API to be all-in or throw (if any of the params is not what we expect, then throw).

Reply:
Hm...I thought we should allow the Register(nullptr, nullptr); - I think what you are all suggesting makes sense. They could fall back to using the Register API with no arguments. It simplifies the design further.

Thanks.